iscsi/multipath.go

```diff
@@ -23,55 +26,57 @@ func ExecWithTimeout(command string, args []string, timeout time.Duration) ([]by
 
 	// This time we can simply use Output() to get the result.
 	out, err := cmd.Output()
-	debug.Println(err)
+	if err != nil {
+		klog.ErrorS(err, "Command error", "command", command, "timeout", timeout, "output", out)
+	}
 
 	// We want to check the context error to see if the timeout was executed.
 	// The error returned by cmd.Output() will be OS specific based on what
 	// happens when a process is killed.
 	if errors.Is(ctx.Err(), context.DeadlineExceeded) {
-		debug.Printf("Command '%s' timeout reached.\n", command)
+		logger.V(1).Info("Command timeout reached", "command", command, "timeout", timeout)
 		return nil, ctx.Err()
 	}
 
 	if err != nil {
 		var ee *exec.ExitError
 		if ok := errors.Is(err, ee); ok {
-			debug.Printf("Non-zero exit code: %s\n", err)
+			klog.ErrorS(err, "Non-zero exit code", "command", command)
 			err = fmt.Errorf("%s", ee.Stderr)
 		}
 	}
 
-	debug.Println("Finished executing command.")
 	return out, err
 }
 
 // FlushMultipathDevice flushes a multipath device dm-x with command multipath -f /dev/dm-x
-func FlushMultipathDevice(device *Device) error {
+func FlushMultipathDevice(logger *logr.Logger, device *Device) error {
```

Comment:
As an exported function, a context would be more future-proof.

Reply:
Switched to using context